packages/cli/src/index.ts

```diff
@@ -62,17 +62,17 @@ const run: (schema: PossibleSchemaInput, options: Partial<ICLIOptions>) => void
 
 const fileName: string | undefined = program.args[0];
 
-if (!process.stdin.isTTY) {
+if (fileName) {
+  const schema: string | object = readFile(fileName);
+  run(schema as PossibleSchemaInput, program as any);
+} else if (!process.stdin.isTTY) {
   let input: string = '';
   process.stdin.resume();
   process.stdin.setEncoding('utf8');
   process.stdin.on('data', (data) => {
     input += data;
   });
   process.stdin.on('end', () => run(safeJSONParse(input) as PossibleSchemaInput, program as any));
```

Comment:
another improvement would be to have an `if` check here to check if `input` is empty and also print an error then.